src/Microsoft.SqlTools.ServiceLayer/DacFx/GenerateDeployPlanOperation.cs

```diff
@@ -0,0 +1,125 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlServer.Dac;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Microsoft.SqlTools.Utility;
+using System;
+using System.Data.SqlClient;
+using System.Diagnostics;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent an in-progress generate deploy plan operation
+    /// </summary>
+    class GenerateDeployPlanOperation : DacFxOperation
+    {
+        public GenerateDeployPlanParams Parameters { get; }
+
+        public GenerateDeployPlanOperation(GenerateDeployPlanParams parameters, ConnectionInfo connInfo): base(connInfo)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        public override void Execute()
```

Comment:
Shouldn't execute do something?

Reply:
The report string needs to be returned to be sent with the result, so the inherited Execute() can't be used.

Reply:
There are probably more elegant ways to deal with this. If the abstraction no longer make sense, let's fix the abstraction. Not hack around it.

Reply:
I changed the abstract method in the base class to be virtual instead. Is this a better way to deal with this?

Reply:
Well, it look like the intent of the Execute() method was to be some sort of command pattern. Given that, I see three other approaches:
1. Do we use the DacFxOperation.Execute() method class directly? If not, and we only deal with concrete classes, then maybe we can delete the Execute() method from the base class altogether? 
2. Execute can set a ReportResults property when complete
3. Enhance Execute() to provide a result, e.g. Execute<T> where the concrete class provides the return type.

Reply:
Ok I added a DeployReport property to the class to store the generated report. Execute() was initially added in the base class so that the code in Execute(TaskExecutionMode mode) can be reused easily.